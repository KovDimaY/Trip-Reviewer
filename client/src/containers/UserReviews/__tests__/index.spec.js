import React from 'react';
import configureStore from 'redux-mock-store';
import { create } from 'react-test-renderer';

import UserPosts from '..';

jest.mock('react-router-dom', () => ({ Link: 'Link' }));
jest.mock('./../../../actions', () => ({
  getUserReviews: jest.fn(() => ({
    type: 'getTripWithReviewer',
  })),
}));

const mockStore = configureStore();

const mockComponent = (initialState = {}, props) => {
  const store = mockStore(initialState);

  return (
    <UserPosts {...props} store={store} />
  );
};

describe('<UserPosts />', () => {
  it('should render component with login', () => {
    const initialState = {
      users: {
        login: {
          id: 'id',
        },
        userPosts: [{
          _id: 'id',
          title: 'title',
          author: 'author',
          createAt: 'createAt',
        }],
      },
    };
    const tree = create(mockComponent(initialState)).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render component with no login', () => {
    const initialState = {
      users: {},
    };
    const tree = create(mockComponent(initialState)).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
