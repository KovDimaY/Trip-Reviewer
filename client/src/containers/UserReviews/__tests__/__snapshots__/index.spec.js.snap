// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserPosts /> should render component with login 1`] = `
<div
  className="user-posts-container limited-width"
>
  <h2>
    Your reviews
  </h2>
  <table
    className="reviews-table"
  >
    <thead>
      <tr>
        <th>
          Title
        </th>
        <th
          className="only-wide-screen"
        >
          Country
        </th>
        <th>
          Date
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td
          className="post-title"
        >
          <Link
            to="/user/edit-post/id"
          >
            title
          </Link>
        </td>
        <td
          className="only-wide-screen"
        >
          country
        </td>
        <td>
          Invalid Date
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`<UserPosts /> should render component with no login 1`] = `
<div
  className="user-posts-container limited-width"
>
  <h2>
    Your reviews
  </h2>
  <table
    className="reviews-table"
  >
    <thead>
      <tr>
        <th>
          Title
        </th>
        <th
          className="only-wide-screen"
        >
          Country
        </th>
        <th>
          Date
        </th>
      </tr>
    </thead>
    <tbody />
  </table>
</div>
`;
